cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -18,80 +18,155 @@ package aws
 
 import (
 	"fmt"
+	"reflect"
 	"sync"
+	"time"
 
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/golang/glog"
 )
 
-type autoScalingGroups struct {
-	registeredAsgs           []*asgInformation
-	instanceToAsg            map[AwsRef]*Asg
-	cacheMutex               sync.Mutex
-	instancesNotInManagedAsg map[AwsRef]struct{}
-	service                  autoScalingWrapper
+const scaleToZeroSupported = false
```

Comment:
Why was scale to zero disabled for AWS?

Reply:
Capturing discussion we had elsewhere: this was totally accidental - sorry!